tests/test_smart_ptr.cpp

```diff
@@ -206,6 +214,18 @@ struct SharedFromThisRef {
     std::shared_ptr<B> shared = std::make_shared<B>();
 };
 
+template <typename T>
+class UniquePtr {
```

Comment:
I would give this a different name to make explicit that one doesn't want to do this for `std::unique_ptr`. E.g. `CustomUniquePtr`.